File: app/Http/Controllers/SpecialistPatternsJava/Creational/AbstractFactory/AbstractFactoryJava/src/windows/Window.java
package windows;

// Интерфейс Window определяет контракт для компонентов окон дома.
public interface Window {
    // Метод open() открывает окно.
    void open();

    // Метод close() закрывает окно.
    void close();

    // Метод install() выполняет установку окна.
    Window install();
}

File: app/Http/Controllers/SpecialistPatterns/AbstractFactory/Windows/Window.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\AbstractFactory\Windows;

// Интерфейс Window определяет контракт для компонентов окон дома.
interface Window {
    // Метод open() открывает окно.
    public function open();

    // Метод close() закрывает окно.
    public function close();

    // Метод install() выполняет установку окна.
    public function install(): Window;
}
